src/SessionStore.php

```diff
@@ -0,0 +1,158 @@
+<?php
+namespace SilverStripe\MFA;
+
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\MFA\Extensions\MemberExtension;
+use SilverStripe\Security\Member;
+
+/**
+ * This class provides an interface to store data in session during an MFA process. This is implemented as a measure to
+ * prevent bleeding state between individual MFA auth types
+ *
+ * @package SilverStripe\MFA
+ */
+class SessionStore
+{
+    const SESSION_KEY = 'thing';
+
+    /**
+     * The member that is currently going through the MFA process
+     *
+     * @var Member
+     */
+    protected $member;
+
+    /**
+     * A string representing the current authentication method that is underway
+     *
+     * @var string
+     */
+    protected $method;
+
+    /**
+     * Any state that the current authentication method needs to retain while it is underway
+     *
+     * @var array
+     */
+    protected $state = [];
+
+    /**
+     * Create a store from the given request getting any initial state from the session of the request
+     *
+     * @param HTTPRequest $request
+     * @return SessionStore
+     */
+    public static function create(HTTPRequest $request)
+    {
+        $state = $request->getSession()->get(static::SESSION_KEY);
+
+        $new = new static;
+
+        if ($state) {
+            $new->setMethod($state['method']);
+            $new->setState($state['state']);
+            if ($state['member']) {
+                $new->setMember(Member::get_by_id($state['member']));
+            }
+        }
+
+        return $new;
```

Comment:
I think you could get rid of this in favour of an injectable service via config and injector DI configuration (I mentioned this above too)